### PR TITLE
User: allow for custom whoami error (fixes #1142)

### LINKED ISSUE #1142: Allow customizing the "whoami" unidentified message
Like `help` can be customized.

> I don't recognize you. You can message me either of these two commands: "user identify <username> <password>" to log in or "user register <username> <password>" to register.

I personally don't give users passwords (`supybot-adduser -u someone -p $(pwgen -s 50) -c \#channel,op`) and just identify them with hostmask or NickServ account and also have disabled registration, so these instructions don't apply to my instance.

I would like to change it to e.g. "You aren't identified. If you think you should be, use the 'auth' command. If you aren't registered and fill the criteria of https://mikaela.info/bot#users, ask Mikaela."

### PR DESCRIPTION
**Pull request by IotaSpencer**
May need to fix some internationalization, but this takes care of a easy feature request.

### CODE REVIEW

**Comment by ProgVal** [comment]
Why do you use `registry.NormalizedString` and not `registry.String`?

Also, instead of having None as a default and have the plugin handle None as a special case, you could put the default message as the default value of `customWho`.
This will fix the crash of test cases, and allow users to see the default string on IRC.

Also, could you rename `customWho` to `customWhoamiError`? I believe that makes its purpose clearer.

File: plugins/User/plugin.py [hunk 1]
```diff
@@ -480,7 +480,7 @@ def whoami(self, irc, msg, args):
             user = ircdb.users.getUser(msg.prefix)
             irc.reply(user.name)
         except KeyError:
-            irc.reply(_('I don\'t recognize you. You can message me either of these two commands: "user identify <username> <password>" to log in or "user register <username> <password>" to register.'))
+            irc.reply(self.registryValue('customWho'))
```
Comment (GLolol) [inline plugins/User/plugin.py]: It has to be `'customWhoamiError'` here as well.

File: plugins/User/plugin.py [hunk 2]
```diff
@@ -480,7 +480,7 @@ def whoami(self, irc, msg, args):
             user = ircdb.users.getUser(msg.prefix)
             irc.reply(user.name)
         except KeyError:
-            irc.reply(self.registryValue('customWho'))
+            irc.reply(self.registryValue('customWhoError'))
```
Comment (ProgVal) [inline plugins/User/plugin.py]: `customWhoamiError`, not `customWhoError`

Comment (IotaSpencer) [inline plugins/User/plugin.py]: Yes just fixed that, lot of stuff on my mind right now.

**Comment by ProgVal** [comment]
One last thing: could you wrap the default value of the config var in `_()`?

**Comment by IotaSpencer** [comment]
there should be all fixed

**Comment by ProgVal** [comment]
Sorry for not seeing it earlier, but the content of the value is not normalized like the help string is, so you have to do it like this:

```
_("I don\'t recognize you. You can message me either of "
    "these two commands: \"user identify <username> <password>\" to log in or "
    "\"user register <username> <password>\" to register.")
```

**Comment by nyuszika7h** [comment]
Nitpick, I'd make it just `whoamiError` because it's not "custom" if it's the default value.

**Comment by ProgVal** [comment]
After giving it some thought, I prefer the original solution, because it makes it possible to alter the message in future versions for existing bots.
I'll cherry-pick your first commit and fix it and stop bothering you with my changes.

**Comment by ProgVal** [comment]
b46ea6f6bced7e10298a6a24506db6e1baceb279

**Pull request closed without merging**